packages/lib/services/plugins/WebviewController.ts

```diff
@@ -49,27 +50,28 @@ export default class WebviewController extends ViewController {
 	private messageListener_: Function = null;
 	private closeResponse_: CloseResponse = null;
 
-	// True if a **panel** is shown in a modal window.
-	private panelInModalMode_ = false;
-
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
 	public constructor(handle: ViewHandle, pluginId: string, store: any, baseDir: string, containerType: ContainerType) {
 		super(handle, pluginId, store);
 		this.baseDir_ = toSystemSlashes(baseDir, 'linux');
 
+		const view: PluginViewState = {
+			id: this.handle,
+			type: this.type,
+			containerType: containerType,
+			html: '',
+			scripts: [],
+			// Opened is used for dialogs and mobile panels (which are shown
+			// like dialogs):
+			opened: containerType === ContainerType.Panel,
```

Comment:
Internally, this pull request works by using `.opened` for panels on mobile. On mobile, `.show` and `.hide` for panels change the `.opened` property in the state. Desktop continues to use `MAIN_LAYOUT_SET_ITEM_PROP` and `appState.mainLayout` to manage plugin panels.

See also the new `private showWithAppLayout() {` method.